Python/testing-sdk/my_project/evaluators/has_substrings.py

```diff
@@ -1,15 +1,23 @@
 import abc
-from typing import Any
+from typing import Generic
 from typing import List
 from typing import Optional
+from typing import TypeVar
 
-from autoblocks.testing.models import BaseTestEvaluator
 from autoblocks.testing.models import BaseTestCase
+from autoblocks.testing.models import BaseTestEvaluator
 from autoblocks.testing.models import Evaluation
 from autoblocks.testing.models import Threshold
 
+TestCaseType = TypeVar("TestCaseType", bound=BaseTestCase)
```

Comment:
this kind of thing would only be necessary if you're making your own ABC, for the most part ppl can ignore stuff like this